File: src/main/java/dk/fitfit/oneliner/service/TagServiceInterface.java
package dk.fitfit.oneliner.service;

import dk.fitfit.oneliner.domain.Tag;

import java.util.List;
import java.util.Set;

public interface TagServiceInterface {
	Set<Tag> findTagsStartingWith(String name);

	Set<Tag> findTagsByRank();

	Tag save(Tag tag);

	List<Tag> findAll();

	Tag getOne(long id);

	long getRank(long id);
}

File: src/main/kotlin/dk/fitfit/oneliner/service/TagServiceInterface.kt
package dk.fitfit.oneliner.service

import dk.fitfit.oneliner.domain.Tag

interface TagServiceInterface {
    fun findTagsStartingWith(name: String): Set<Tag>

    fun findTagsByRank(): Set<Tag>

    fun save(tag: Tag): Tag

    fun findAll(): List<Tag>

    fun getOne(id: Long): Tag

    fun getRank(id: Long): Long

    fun fuzzySearchDescription(searchString: String): MutableList<*>
}
